pam/authmodeselection.go

```diff
@@ -241,17 +243,16 @@ func getAuthenticationModes(client authd.PAMClient, sessionID string, uiLayouts
 
 		gamResp, err := client.GetAuthenticationModes(context.Background(), gamReq)
 		if err != nil {
-			return pamSystemError{
-				msg: fmt.Sprintf("could not get authentication modes: %v", err),
-			}
+			return pam.NewTransactionError(pam.ErrSystem,
+				fmt.Errorf("could not get authentication modes: %v", err))
 		}
 
 		authModes := gamResp.GetAuthenticationModes()
 		if len(authModes) == 0 {
-			return pamIgnore{
+			return newPamIgnore(
 				// TODO: probably go back to broker selection here
-				msg: "no supported authentication mode available for this provider",
-			}
+				"",
```

Comment:
The fact that you need to set an empty field here is again a good sign that something is not idiomatic in this wrapping :)

Reply:
Well, we can indeed use `pamIgnore{}` in this case.